### PR TITLE
Brand new logger

### PR DESCRIPTION
**Pull request by lvivier**
ref retsly/retsly#1039

We're rewriting the logger from scratch, with new features and a new architecture base on [segmentio/yal](/segmentio/yal).

## TODO

- [x] Add log server
- [x] Console plugin
- [x] HipChat plugin
- [x] LogEntries plugin
- [x] New logging client
- [x] Logging client works with browserify
- [x] Logging client works with component
- [x] Commander options for `logd`
- [x] Documentation

### Client-side

Can use xhr or jsonp, whatever's needed to work with old IE. Has exactly the same interface, but `addrs` supplied to the constructor will be an http url like `https://logs.rets.ly/log` or w/e

### Express server

Need a CORS header, improved validation, improved error handling. Authentication/rate limiting?

### Documentation

How to set up `logd`, how to use log client in projects.

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+      critical: 'red',
+      alert: 'red',
+      emergency: 'red'
+    }
+
+    yal.on('message', function(msg){
+      //TODO fix notify
+      var notify = (msg.level) ? true : false
+      var color = colorMap[msg.level] || 'grey'
```
Comment (lvivier) [inline lib/server.js]: `gray`

File: package.json [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
+    "axon": "~1.0.0",
+    "body-parser": "^1.10.2",
     "cli-color": "^0.3.2",
+    "express": "^4.11.1",
     "format": "^0.2.1",
-    "superagent": "^0.20.0"
+    "prettyjson": "^1.0.0",
+    "superagent": "^0.21.0",
+    "xhr": "^2.0.1",
+    "yal-hipchat": "^0.1.0",
```
Comment (lvivier) [inline package.json]: can remove this

File: bin/logd [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ */
+
+var PORT = process.env.LOG_SERVER_PORT || 57000
+
+
+/**
+ * HTTP server port
+ */
+
+var HTTP_PORT = process.env.LOG_HTTP_PORT || 3000
```
Comment (lvivier) [inline bin/logd]: @switters I moved the http server to its own port and removed the default hipchat token

**Comment by switters** [comment]
@lvivier - this PRs getting too large. Pull and create issues for whatever else is on the wishlist?

**Comment by lvivier** [comment]
@switters yeah, I'll merge this and we can open some issues

**Pull request merged**